torchtune/utils/_generation.py

```diff
@@ -68,6 +68,7 @@ def generate(
     temperature: float = 1.0,
     top_k: Optional[int] = None,
     eos_id: Optional[int] = None,
+    custom_generate_next_token: Optional[Callable] = None,
```

Comment:
Update the doc string below please. And explicitly mention that this is used for the case where we compile the function. 

From a design perspective, I wonder if its better to just send a boolean flag called ```compile``` to this function and if so, just compile the ```generate_next_token``` here? Isn't this where it's actually called?

Reply:
I started with the boolean flag, but since we are timing this section, we can't do compilation inside this function (and also the prefill run), that's why I moved the compilation outside

Reply:
ok yeh that makes sense - thanks for the clarification!